### PR TITLE
Feat/85290 admin list users & Lock users

### PR DESCRIPTION
**Pull request by assimoes-bjss**
- Adds Service Roles validator to admin endpoints
- Creates List users endpoint (search by type or email)
- Reworks Lock User endpoint

### CODE REVIEW

File: __external__/services/src/__tests__/Admin.service.spec.ts [hunk 1]
```diff
@@ -70,12 +72,13 @@ describe("[User Account Lock suite", () => {
...
+    const result = await adminService.userLockValidation(assessmentUser.id);
 
-    expect(result.length).toBeGreaterThan(0);
-    expect(result[0].error.code).toBe("LastAssessmentUserOnPlatformError");
+    expect(result).toBeDefined();
+    expect(result.lastAssessmentUserOnPlatform.valid).toBe(false);
+    expect(result.lastAccessorUserOnOrganisation.valid).toBe(true);
+    expect(result.lastAccessorUserOnOrganisationUnit.valid).toBe(true);
+    expect(result.lastAccessorFromUnitProvidingSupport.valid).toBe(true);
   });
```
Comment (ShaliniNHS) [inline __external__/services/src/__tests__/Admin.service.spec.ts:82]: We should avoid having multiple asserts, like this test is for last assessment user so doesn't make sense validating accessor users result.

File: adminsLockUsers/index.ts [hunk 2]
```diff
@@ -1,46 +1,38 @@
...
 import * as persistence from "./persistence";
-import * as validation from "./validation";
 
 class AdminsLockUsers {
   @AppInsights()
   @SQLConnector()
-  @Validator(validation.ValidatePayload, "body", "Invalid Payload")
-  @JwtDecoder()
+  @JwtDecoder(true)
+  @ServiceRoleValidator(ServiceRole.ADMIN, ServiceRole.SERVICE_TEAM)
```
Comment (ShaliniNHS) [inline adminsLockUsers/index.ts:19]: Shouldn't we use the AllowUserType decorator before ServiceRoleValidator? That will take care of validating the type & setting up the request user in context, which right now is being done in the function & service.

**Review by ShaliniNHS — COMMENTED** [review]
Newly added function specs need to be added to Open API

File: adminsLockUsers/index.ts (hunk 2 above)
Comment (assimoes-bjss) [reply adminsLockUsers/index.ts:19]: No. The request user is injected in the JwtDecoder decorator

File: __external__/services/src/__tests__/Admin.service.spec.ts (hunk 1 above)
Comment (assimoes-bjss) [reply __external__/services/src/__tests__/Admin.service.spec.ts:82]: This test depends on the combination of all these values. I can entertain the possibility of creating an object with the response and compare the result with that expected object. However, that makes the assertion less readable.

Comment (assimoes-bjss) [reply __external__/services/src/__tests__/Admin.service.spec.ts:82]: image

Comment (assimoes-bjss) [reply __external__/services/src/__tests__/Admin.service.spec.ts:82]: If I were to use an object comparison, I will not be immediatly evident which depedent rule is failing.

Comment (ShaliniNHS) [reply __external__/services/src/__tests__/Admin.service.spec.ts:82]: thanks, saw the dependency on the fields

File: adminsLockUsers/index.ts (hunk 2 above)
Comment (assimoes-bjss) [reply adminsLockUsers/index.ts:19]: Yep. This is correct, I will update the admin endpoints to include this decorator. But that decorator is only validating the type of user. The request user is injected on the `decodeJwt` decorator. Still, it's a good catch.

File: __external__/services/src/__tests__/Admin.service.spec.ts (hunk 1 above)
Comment (assimoes-bjss) [reply __external__/services/src/__tests__/Admin.service.spec.ts:82]: Still nice to point this out. But really in this case I think this is the exception regarding multiple asserts.

**Comment by assimoes-bjss** [comment]
> 

Yep. This is still a work in progress. Not merging yet.

**Review by ShaliniNHS — APPROVED** [review]

**Review by JyotiJaiswal169 — APPROVED** [review]

**Pull request merged**